### PR TITLE
Enable Intercom in mobile apps

### LINKED ISSUE #3326: intercom is not working in mobile
intercom has a whitelist and the URL gets obscured in the native apps so they cannot be whitelisted. i have contacted support and they recommended the intercom / ionic docs.

there is a way to install intercom for mobile apps with ionic that should be researched.

### PR DESCRIPTION
**Pull request by mariopino**
Closes #3326 

**Description:**

Enable Intercom in mobile apps.

Implemented using https://github.com/stewwan/capacitor-intercom. Currently to test this pr you should apply this pr manually in the module: https://github.com/stewwan/capacitor-intercom/pull/9.

image

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: android/app/src/main/assets/capacitor.config.json [hunk 1]
```diff
@@ -3,5 +3,13 @@
   "appName": "Lunie",
   "bundledWebRuntime": false,
   "npmClient": "yarn",
-  "webDir": "dist"
+  "webDir": "dist",
+  "plugins": {
+    "IntercomPlugin": {
+      "android-apiKey": "android_sdk-xxxx",
+      "android-appId": "yyyy",
+      "ios-apiKey": "ios_sdk-xxxx",
```
Comment (faboweb) [inline android/app/src/main/assets/capacitor.config.json]: this is for android only, so this needs to move into the ios config

File: capacitor.config.json [hunk 2]
```diff
@@ -3,5 +3,13 @@
   "appName": "Lunie",
   "bundledWebRuntime": false,
   "npmClient": "yarn",
-  "webDir": "dist"
+  "webDir": "dist",
+  "plugins": {
+    "IntercomPlugin": {
+      "android-apiKey": "android_sdk-xxxx",
+      "android-appId": "yyyy",
```
Comment (faboweb) [inline capacitor.config.json]: or remove the other keys appropriate ^^

File: src/main.js [hunk 3]
```diff
@@ -43,6 +44,14 @@ const urlParams = getURLParams(window)
 const { store, router, apolloProvider } = init(urlParams)
 const { SplashScreen, StatusBar } = Plugins
 
+// Mobile Intercom
+if (config.mobileApp) {
+  const intercom = new Intercom()
+  const userId = "lunie-app-" + Math.floor(Math.random() * 10000 + 1).toString()
+  intercom.registerIdentifiedUser({ userId })
+  Vue.prototype.$mobileIntercom = intercom
```
Comment (faboweb) [inline src/main.js]: ooooook, better would have been to pass it to the store and trigger an event from the component to do something in the store. but this probably works

Comment (mariopino) [reply src/main.js]: I can do it!

File: android/app/src/main/assets/capacitor.config.json [hunk 4]
```diff
@@ -3,5 +3,11 @@
   "appName": "Lunie",
   "bundledWebRuntime": false,
   "npmClient": "yarn",
-  "webDir": "dist"
+  "webDir": "dist",
+  "plugins": {
+    "IntercomPlugin": {
+      "android-apiKey": "android_sdk-xxxx",
+      "android-appId": "yyyy"
```
Comment (jbibla) [inline android/app/src/main/assets/capacitor.config.json]: should be null / empty string? probably doesn't matter.

**Comment by jbibla** [comment]
is `TmDataMsg` actually used anymore? if not, we can delete it. 

is the intercom button / link somewhere else? i feel like it is...

missing unit tests for: https://codecov.io/gh/luniehq/lunie/compare/5eabe7d8716a357eff9a09be4e44a5dacf75e0ae...d4c6eb901e908c92210f0a64672becf360849a69/src/src/components/common/TmConnectedNetwork.vue#L102

**Comment by faboweb** [comment]
> is TmDataMsg actually used anymore? if not, we can delete it.

used in several places

**Comment by jbibla** [comment]
sorry, i meant `TmDataError`

**Comment by faboweb** [comment]
> TmDataError

used in several places i.e. TmPage

**Comment by jbibla** [comment]
i know it's _used_ but does it ever get displayed? the message is "even though you're connected to a node ..."

**Comment by mariopino** [comment]
Hey @jbibla @faboweb what do you guys think it's best? Implement intercom with store or simply finish testing and doing it in other pr.

I think implementing intercom stuff in store is better, of course, we can add actions in the future to do some interesting things like directly open the HelpCenter. In this PR only `registerIdentifiedUser()` and `displayMessenger()` actions would be implemented (implementing the other calls is straightforward thought).

File: android/app/src/main/assets/capacitor.config.json (hunk 4 above)
Comment (mariopino) [reply android/app/src/main/assets/capacitor.config.json]: Yep, before app building should be replaced by the correct values by the script Fabo coded yesterday.

**Comment by faboweb** [comment]
@jbibla the message is wrong that we should fix. but if a gql query fails, in some cases it should show right?

**Comment by faboweb** [comment]
Merging this PR
@mariopino will create a follow up with improvements to use the store

**Pull request merged**